zerver/views/message_fetch.py

```diff
@@ -1135,11 +1140,9 @@ def limit_query_to_range(query: Query,
         # for something like `message_id = 42` is exactly what we want.  In other
         # cases, which could possibly be buggy API clients, at least we will
         # return at most one row here.
-        query = query.where(id_col == anchor)
-
-    return query
+        return query.where(id_col == anchor)
```

Comment:
Can you do this structural refactor to have each clause return as a prep commit?